### ISSUE TITLE
False positive in "declaration-empty-line-before" caused by shared-line comments

### ISSUE THREAD

**Issue by nwoltman** [stylelint/stylelint#2824]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where shared-line comments cause `declaration-empty-line-before` to report errors where it shouldn't.

> Which rule, if any, is this issue related to?

[`declaration-empty-line-before`](https://stylelint.io/user-guide/rules/declaration-empty-line-before/)

> What CSS is needed to reproduce this issue?

```css
.foo { /* shared-line comment */
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "declaration-empty-line-before": ["always", {
      "except": ["first-nested"]
    }],
  }
}
```

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

grunt-stylelint

> Does your issue relate to non-standard syntax?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened?

The following warnings were flagged:

```shell
test.css
 2:3  ×  Expected empty line before declaration   declaration-empty-line-before
```
---

Related to #2764.
Issue probably introduced in #2262.

**Comment by jeddy3** [comment]
@nwoltman Thanks for the report.

> Related to #2764.

Yes, you're right. It looks like the bug is applicable to more than just the `at-rule-empty-line-before` rule. I've updated the title of the other issue accordingly so we can merge this into that one. The fix will be the same across all `*-empty-line-before` rules. @nwoltman Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) for https://github.com/stylelint/stylelint/issues/2764 if you'd like to see this issue resolved. I suspect improving the [`isFirstNested` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/isFirstNested.js#L10) to account for shared line comments should do it.
